python/usb_console/run_simulation.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/local/bin/python3
+
+from argparse import ArgumentParser
+from state_session import StateSession
+from cmdprompt import StateCmdPrompt
+from data_containers import Logger, Datastore
+from simulation import Simulation
+import json, sys, os, threading, time
+
+class SimulationRun(object):
+    def __init__(self, random_seed, data_dir, device_data):
+        self.random_seed = random_seed
+        self.data_dir = data_dir
+        self.device_data = device_data
+
+        self.datastores = {}
+        self.loggers = {}
+        self.devices = {}
+
+    def start(self):
+        """Starts a run of the simulation."""
+
+        if not device_data:
+            print('Error: must specify at least one serial port.')
+            raise SystemExit
+
+        # Set up test table by connecting to each device specified in the config.
+        for device in self.device_data:
+            device_name = device["name"]
+
+            # Create directory for run data
+            simulation_run_dir = os.path.join(self.data_dir, time.strftime("%Y%m%d-%H%M%S"))
+            os.makedirs(simulation_run_dir, exist_ok=True)
+
+            # Generate data loggers and device manager for device
+            device_datastore = Datastore(device_name, simulation_run_dir)
+            device_logger = Logger(device_name, simulation_run_dir)
+            port_cmd = StateSession(self.data_dir, device_name, device_datastore, device_logger)
+
+            # Connect to device, failing gracefully if device connection fails
+            if port_cmd.connect(device["port"], 115200):
+                self.devices[device_name] = port_cmd
+                self.datastores[device_name] = device_datastore
+                self.loggers[device_name] = device_logger
+
+                device_datastore.start()
+                device_logger.start()
+            elif device["stop_if_unconnected"]:
+                print("Error: exiting due to a required serial port being disconnected.")
+                self.stop_all()
+                return
+
+        # Set up MATLAB simulation
+        self.sim = Simulation(self.devices, self.random_seed)
+        self.sim_thread = threading.Thread(name="Python-MATLAB Simulation Interface", target=self.sim.run, args=[2])
+        self.sim_thread.start()
```

Comment:
We may want to move the thread spawning into the simulation object. Just because when we actually sim two satellites we will have two MATLAB instances running in parallel which will give us much better performance.

Reply:
We're probably going to be bottle necked by communication not computation.

Reply:
Got it, I'll address this.

Reply:
Oh really? Okay. Wasn't aware we had the sim moving that much faster. We can discuss com time vs comp time in slack or something as well.